File: Console-XHSTT/models/ItalyInstance1/events.py
from typing import List, Union, Optional
from enum import Enum


class TimeElement:
    reference: str

    def __init__(self, reference: str) -> None:
        self.reference = reference


class EventEventGroups:
    event_group: Union[List[TimeElement], TimeElement]

    def __init__(self, event_group: Union[List[TimeElement], TimeElement]) -> None:
        self.event_group = event_group


class Role(Enum):
    CLASS = "Class"
    CLASS1 = "Class1"
    COPRESENCE = "Copresence"
    CO_PRESENCE = "CoPresence"
    TEACHER = "Teacher"


class ResourceElement:
    role: Role
    resource_type: TimeElement
    reference: str

    def __init__(self, role: Role, resource_type: TimeElement, reference: str) -> None:
        self.role = role
        self.resource_type = resource_type
        self.reference = reference


class Resources:
    resource: Union[List[ResourceElement], ResourceElement]

    def __init__(self, resource: Union[List[ResourceElement], ResourceElement]) -> None:
        self.resource = resource


class Event:
    name: str
    duration: int
    course: TimeElement
    resources: Resources
    event_groups: Optional[EventEventGroups]
    id: str
    time: Optional[TimeElement]

    def __init__(self, name: str, duration: int, course: TimeElement, resources: Resources, event_groups: Optional[EventEventGroups], id: str, time: Optional[TimeElement]) -> None:
        self.name = name
        self.duration = duration
        self.course = course
        self.resources = resources
        self.event_groups = event_groups
        self.id = id
        self.time = time


class EventGroupsCourse:
    name: str
    id: str

    def __init__(self, name: str, id: str) -> None:
        self.name = name
        self.id = id


class EventsEventGroups:
    course: List[EventGroupsCourse]
    event_group: List[EventGroupsCourse]

    def __init__(self, course: List[EventGroupsCourse], event_group: List[EventGroupsCourse]) -> None:
        self.course = course
        self.event_group = event_group


class Events:
    event_groups: EventsEventGroups
    event: List[Event]

    def __init__(self, event_groups: EventsEventGroups, event: List[Event]) -> None:
        self.event_groups = event_groups
        self.event = event

File: Web-XHSTT/sakai-ng/src/app/api/events.ts
export interface Events {
    EventGroups: EventGroups
    Event: Event[]
  }

  export interface EventGroups {
    Course: Course[]
    EventGroup: EventGroup
  }

  export interface Course {
    Id: string
    Name: string
  }

  export interface EventGroup {
    Id: string
    Name: string
  }

  export interface Event {
    Id: string
    Name: string
    Duration: string
    Course: Course2
    Resources: Resource[]
    EventGroups: EventGroups2
  }

  export interface Course2 {
    "@Reference": string
  }

  export interface Resource {
    "@Reference": string
    Role: string
    ResourceType: ResourceType
  }

  export interface ResourceType {
    "@Reference": string
  }

  export interface EventGroups2 {
    EventGroup: EventGroup2
  }

  export interface EventGroup2 {
    "@Reference": string
  }
